core/storage.js

```diff
@@ -52,17 +26,17 @@ define('core/storage',
             }
         },
         removeItem: function(key) {
-            ls.removeItem(mapKey(key));
+            localStorage.removeItem(mapKey(key));
         },
         setItem: function(key, value) {
             try {
-                ls.setItem(mapKey(key), JSON.stringify(value));
+                localStorage.setItem(mapKey(key), JSON.stringify(value));
             } catch (e) {
                 // Clear localStorage if the quota was reached.
                 if (e.name == 'QuotaExceededError' ||
                     e.name == 'NS_ERROR_DOM_QUOTA_REACHED') {
-                    console.log('LocalStorage full, clearing and reloading');
-                    ls.clear();
+                    require('core/log')('storage').log('LS full, clearing');
```

Comment:
It might be nice to save the user token and put it back after we call`localStorage.clear()`. Not sure if that's worth doing though.

This also seems like something we should track. I think the tracking module is outside of core so we'd need to emit an event for that to pick up. I don't think we have any way of observing objects so I guess `z.page.trigger('storage-cleared', {reason: 'full'})` or something?